ddl/placement_policy_test.go

```diff
@@ -534,3 +534,124 @@ func (s *testDBSuite6) TestPolicyCacheAndPolicyDependencyCache(c *C) {
 	c.Assert(dependencies, NotNil)
 	c.Assert(len(dependencies), Equals, 0)
 }
+
+func (s *testDBSuite6) TestAlterTablePartitionWithPlacementPolicy(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	defer func() {
+		tk.MustExec("drop table if exists t1")
+		tk.MustExec("drop placement policy if exists x")
+	}()
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t1")
+	// Direct placement option: special constraints may be incompatible with common constraint.
+	tk.MustExec("create table t1 (c int) PARTITION BY RANGE (c) " +
+		"(PARTITION p0 VALUES LESS THAN (6)," +
+		"PARTITION p1 VALUES LESS THAN (11)," +
+		"PARTITION p2 VALUES LESS THAN (16)," +
+		"PARTITION p3 VALUES LESS THAN (21));")
+
+	tk.MustExec("alter table t1 partition p0 " +
+		"PRIMARY_REGION=\"cn-east-1\" " +
+		"REGIONS=\"cn-east-1, cn-east-2\" " +
+		"FOLLOWERS=2 " +
+		"FOLLOWER_CONSTRAINTS=\"[+zone=cn-east-1]\" " +
+		"CONSTRAINTS=\"[+disk=ssd]\"")
+
+	tbl := testGetTableByName(c, tk.Se, "test", "t1")
+	c.Assert(tbl, NotNil)
+	ptDef := testGetPartitionDefinitionsByName(c, tk.Se, "test", "t1", "p0")
+	c.Assert(ptDef.PlacementPolicyRef.Name.L, Equals, "")
```

Comment:
Should `ptDef.PlacementPolicyRef` be nil here?

Reply:
the args of json.Unmarshal() will be init , and it seems to me that ptDef.PlacementPolicyRef is nil or "" makes no difference